org.civicrm.reqangular/src/common/decorators/uib-tooltip.decorator.js

```diff
@@ -0,0 +1,150 @@
+           */
+          function makeTooltipClickable (attributes) {
+            isTouchDevice && injectOverlayToTriggerElement();
+            setEventHandlersToTriggerElements();
+          }
+
+          /**
+           * Sets handlers to the tooltip trigger elements
+           */
+          function setEventHandlersToTriggerElements () {
+            if (isTouchDevice) {
+              $overlay.on('touchend', function (event) {
+                toggleTooltip('source', true, false, event);
+                !scope.open && setEventHandlersToTooltip();
+              });
+              $document.find('body').on('touchend', function () {
+                toggleTooltip('source', false, false);
+                toggleTooltip('tooltip', false, false);
+              });
+              $overlay.on('click', function (event) {
+                event.stopPropagation();
+              });
+            } else {
+              $element.on('mouseenter', function () {
+                toggleTooltip('source', true, false);
+                !scope.open && setEventHandlersToTooltip();
+              });
+              $element.on('mouseleave', function () {
+                toggleTooltip('source', false, true);
+              });
+            }
+
+            $element.on(clickType, function () {
+              toggleTooltip('source', false, false);
+              toggleTooltip('tooltip', false, false);
+            });
+          }
+
+          /**
+           * Sets event handlers to the tooltip itself
+           */
+          function setEventHandlersToTooltip () {
+            var $tooltip;
+
+            $timeout(function () {
+              $tooltip = $document.find('.tooltip-clickable-template:visible:last');
+
+              $tooltip.on('mouseenter', function () {
+                toggleTooltip('tooltip', true, false);
+              });
+              $tooltip.on('mouseleave', function () {
+                toggleTooltip('tooltip', false, true);
+              });
+              $tooltip.on(clickType, function () {
+                toggleTooltip('source', false, false);
+                toggleTooltip('tooltip', false, false);
+              });
+            });
+          }
+
+          /**
+           * Sets initial attributes for clickable tooltip
+           */
+          function setInitialAttributesForClickableTooltip () {
+            attributes.tooltipTrigger = 'customShow';
+            attributes.tooltipAnimation = false;
+          }
+
+          /**
+           * Shows/hides tooltip
+           *
+           * @param {String} elementType source|tooltip
+           * @param {Boolean} elementIsHovered is element currently hovered or not
+           * @param {Boolean} delayEvent should the event be delayed
+           * @param {Event} eventToNotBePropagated if passed, it will not be propagated
+           */
+          function toggleTooltip (elementType, elementIsHovered, delayEvent, eventToNotBePropagated) {
```

Comment:
Why don't you make the delayEvent a number instead of a boolean to help a bit with the readability?

```js
toggleTooltip('source', true, 100); // with a nice default of 0
```
It's a bit difficult to infer what `source, true, true` means, but `source, true, 100` shades a bit more of a clue. Using an object might shade even more light though `{element: 'source', hover: false, delay: 100}` (up to you).

Reply:
Done.